### PR TITLE
Adding SystemFormats.Nrbf fixes

### PR DESCRIPTION
**Pull request by pchaurasia14**
Pull in changes from #9465 and #9492 to support nrbf code for Preview7

## Customer Impact
Customers using .NET 9 Preview 7 would continue to take in implicit dependency on BinaryFormatter nuget package.

## Regression
No. 

## Testing
Local testing and compat-lab applications

## Risk
Medium.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/wpf/pull/9506)

cc - @adamsitnik , @ericstj

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/tests/UnitTests/PresentationCore.Tests/PresentationCore.Tests.csproj [hunk 1]
```diff
@@ -25,6 +25,8 @@
     <PackageReference Include="FluentAssertions" Version="$(FluentAssertionsVersion)" />
     <PackageReference Include="System.Collections.Concurrent" Version="4.3.0" />
     <PackageReference Include="System.Runtime.Serialization.Formatters" Version="9.0.0-preview.7.24359.11 " />
```
Comment (ericstj) [inline src/Microsoft.DotNet.Wpf/tests/UnitTests/PresentationCore.Tests/PresentationCore.Tests.csproj:27]: This looks wrong.  We shouldn't have hardcoded versions in the projects.

Comment (ericstj) [reply src/Microsoft.DotNet.Wpf/tests/UnitTests/PresentationCore.Tests/PresentationCore.Tests.csproj:27]: I see, that was fixed with https://github.com/dotnet/wpf/commit/fbff6e12cdc0bac3256e171dcd00bb95319b9485

Comment (pchaurasia14) [reply src/Microsoft.DotNet.Wpf/tests/UnitTests/PresentationCore.Tests/PresentationCore.Tests.csproj:27]: I tried to keep the changes to minimum and pick only the changes that were introduced in the 2 original PRs.

**Comment by pchaurasia14** [comment]
Closing in favor of #9507 .

**Pull request closed without merging**